app/models/concerns/applicable_to_nations.rb

```diff
@@ -0,0 +1,51 @@
+require 'uri'
+
+module ApplicableToNations
+  extend ActiveSupport::Concern
+
+  included do
+    validate :applicable_to_at_least_one_nation
+    validate :alternative_urls_are_valid
+  end
+
+  def possible_nations
+    [
+      :england,
+      :northern_ireland,
+      :scotland,
+      :wales,
+    ]
+  end
+
+  def applicable_nations
+    applicable_nations = possible_nations.select { |n|
+      self.send(n) == true
+    }
+  end
+
+  def inapplicable_nations
+    possible_nations - applicable_nations
+  end
+
+private
+
+  def applicable_to_at_least_one_nation
+    if applicable_nations.empty?
+      errors.add(:applicability, "must have at least one nation")
+    end
+  end
+
+  def alternative_urls_are_valid
+    alt_policy_urls = inapplicable_nations.select { |nation|
+      self.send(:"#{nation}_policy_url").present?
+    }
+
+    nations_with_bad_urls = alt_policy_urls.reject { |nation|
+      self.send(:"#{nation}_policy_url") =~ URI::regexp
```

Comment:
This is way too loose. It will validate `"ftp://foo.com"`, `"urn:isbn:0451450523"` as well as `"htt://typo.com"` and `"Here is a link! http://website.com"`. I suggest either use a slightly tweaked regular expression like `/\A#{URI::regexp(['http', 'https'])}\z/`, or follow the code in the whitehall validator and use `Addressable` to parse the URI.